dpctl/tensor/_copy_utils.py

```diff
@@ -279,7 +279,10 @@ def _copy_from_usm_ndarray_to_usm_ndarray(dst, src):
         common_shape = common_shape[ones_count:]
 
     if src.ndim < len(common_shape):
-        new_src_strides = (0,) * (len(common_shape) - src.ndim) + src.strides
+        pad_count = len(common_shape) - src.ndim
+        new_src_strides = (0,) * pad_count + tuple(
+            s if d > 1 else 0 for s, d in zip(src.strides, src.shape)
+        )
```

Comment:
I wonder, is there a reason not to reuse the ``_broadcast_strides`` function from ``_manipulation_functions`` here?

I tested it out and (besides some importing issues) it seemed to give the same results, passed tests, etc.

Reply:
I moved the implementation of `_broadcast_strides` to `_copy_utils` and imported it from `_manupulation_functions.py`.